File: Exercism/csharp/bob/Bob.cs
using System;
using System.Linq;

public static class Bob
{
	public static string Response(string input)
	{
		// get rid of whitespaces at the end of the string
		input.Trim(); // this is crap
		input.Replace(" ", ""); // this is crap
		input.Replace("\t", "");
		input.Replace("\n", "");

		string s = new string(input.Where(c => !char.IsWhiteSpace(c)).ToArray<char>());

		// check if characters are uppercased (shouting)
		bool shouting = true;
		foreach(char c in s)
		{
			if(Char.IsLower(c) && !Char.IsDigit(c) && !Char.IsPunctuation(c))
			{
				shouting = false;
				break;
			}
		}

		// check if string contains numbers and punctuals only
		bool numsOnly = true;
		foreach(char c in s)
		{
			if(!Char.IsDigit(c) && !Char.IsPunctuation(c) && !Char.IsWhiteSpace(c))
			{
				numsOnly = false;
				break;
			}
		}

		if(shouting && s.Length > 0 && !numsOnly && s.EndsWith('?'))
		{
			return "Calm down, I know what I'm doing!";
		}
		else if(shouting && s.Length > 0 && !numsOnly)
		{
			return "Whoa, chill out!";
		}
		else if(s.EndsWith('?'))
		{
			return "Sure.";
		}
		else if(s.Length == 0)
		{
			return "Fine. Be that way!";
		}

		return "Whatever.";
	}
}

File: Exercism/typescript/bob/bob.ts
export function hey(_s: string): string {
  let trimmed = _s.trim()
  let empty = trimmed.length === 0
  let question = trimmed.endsWith('?')
  let shouting = /[A-Z]/.test(trimmed) && !/[a-z]/.test(trimmed)

  if(empty) {
    return "Fine. Be that way!"
  } else if(shouting && question) {
    return "Calm down, I know what I'm doing!"
  } else if(question) {
    return "Sure."
  } else if(shouting) {
    return "Whoa, chill out!"
  }

  return "Whatever.";
}
